opal/threads/thread_usage.h

```diff
@@ -1,266 +1,29 @@
-#define OPAL_ATOMIC_FETCH_SUB_SIZE_T opal_thread_fetch_sub_size_t
-
-#define OPAL_THREAD_COMPARE_EXCHANGE_STRONG_32 opal_thread_compare_exchange_strong_32
-#define OPAL_ATOMIC_COMPARE_EXCHANGE_STRONG_32 opal_thread_compare_exchange_strong_32
-
-#define OPAL_THREAD_COMPARE_EXCHANGE_STRONG_PTR(x, y, z) opal_thread_compare_exchange_strong_ptr ((opal_atomic_intptr_t *) x, (intptr_t *) y, (intptr_t) z)
-#define OPAL_ATOMIC_COMPARE_EXCHANGE_STRONG_PTR OPAL_THREAD_COMPARE_EXCHANGE_STRONG_PTR
-
-#define OPAL_THREAD_SWAP_32 opal_thread_swap_32
-#define OPAL_ATOMIC_SWAP_32 opal_thread_swap_32
-
-#define OPAL_THREAD_SWAP_PTR(x, y) opal_thread_swap_ptr ((opal_atomic_intptr_t *) x, (intptr_t) y)
-#define OPAL_ATOMIC_SWAP_PTR OPAL_THREAD_SWAP_PTR
-
-/* define 64-bit macros is 64-bit atomic math is available */
-#if OPAL_HAVE_ATOMIC_MATH_64
-
-OPAL_THREAD_DEFINE_ATOMIC_OP(int64_t, add, +, 64)
-OPAL_THREAD_DEFINE_ATOMIC_OP(int64_t, and, &, 64)
-OPAL_THREAD_DEFINE_ATOMIC_OP(int64_t, or, |, 64)
-OPAL_THREAD_DEFINE_ATOMIC_OP(int64_t, xor, ^, 64)
-OPAL_THREAD_DEFINE_ATOMIC_OP(int64_t, sub, -, 64)
-OPAL_THREAD_DEFINE_ATOMIC_COMPARE_EXCHANGE(int64_t, int64_t, 64)
-OPAL_THREAD_DEFINE_ATOMIC_SWAP(int64_t, int64_t, 64)
-
-#define OPAL_THREAD_ADD_FETCH64 opal_thread_add_fetch_64
-#define OPAL_ATOMIC_ADD_FETCH64 opal_thread_add_fetch_64
-
-#define OPAL_THREAD_AND_FETCH64 opal_thread_and_fetch_64
-#define OPAL_ATOMIC_AND_FETCH64 opal_thread_and_fetch_64
-
-#define OPAL_THREAD_OR_FETCH64 opal_thread_or_fetch_64
-#define OPAL_ATOMIC_OR_FETCH64 opal_thread_or_fetch_64
-
-#define OPAL_THREAD_XOR_FETCH64 opal_thread_xor_fetch_64
-#define OPAL_ATOMIC_XOR_FETCH64 opal_thread_xor_fetch_64
-
-#define OPAL_THREAD_FETCH_ADD64 opal_thread_fetch_add_64
-#define OPAL_ATOMIC_FETCH_ADD64 opal_thread_fetch_add_64
-
-#define OPAL_THREAD_FETCH_AND64 opal_thread_fetch_and_64
-#define OPAL_ATOMIC_FETCH_AND64 opal_thread_fetch_and_64
-
-#define OPAL_THREAD_FETCH_OR64 opal_thread_fetch_or_64
-#define OPAL_ATOMIC_FETCH_OR64 opal_thread_fetch_or_64
-
-#define OPAL_THREAD_FETCH_XOR64 opal_thread_fetch_xor_64
-#define OPAL_ATOMIC_FETCH_XOR64 opal_thread_fetch_xor_64
-
-#define OPAL_THREAD_COMPARE_EXCHANGE_STRONG_64 opal_thread_compare_exchange_strong_64
-#define OPAL_ATOMIC_COMPARE_EXCHANGE_STRONG_64 opal_thread_compare_exchange_strong_64
-
-#define OPAL_THREAD_SWAP_64 opal_thread_swap_64
-#define OPAL_ATOMIC_SWAP_64 opal_thread_swap_64
-
-#endif
-
-/* thread local storage */
-#if OPAL_C_HAVE__THREAD_LOCAL
-#define opal_thread_local _Thread_local
-#define OPAL_HAVE_THREAD_LOCAL 1
-
-#elif OPAL_C_HAVE___THREAD /* OPAL_C_HAVE__THREAD_LOCAL */
-#define opal_thread_local __thread
-#define OPAL_HAVE_THREAD_LOCAL 1
-#endif /* OPAL_C_HAVE___THREAD */
+#ifndef OPAL_THREAD_USAGE_H
+#define OPAL_THREAD_USAGE_H 1
 
-#if !defined(OPAL_HAVE_THREAD_LOCAL)
-#define OPAL_HAVE_THREAD_LOCAL 0
-#endif /* !defined(OPAL_HAVE_THREAD_LOCAL) */
+#include "opal/mca/threads/thread_usage.h"
```

Comment:
Ditto with `opal/threads/condition.h ` -- it looks like this file should be removed.